src/python/pants/backend/jvm/subsystems/zinc.py

```diff
@@ -212,37 +213,30 @@ def native_image(self, context):
     return self._zinc_factory.hackily_snapshot(context)
 
   @memoized_property
-  def dist(self):
-    """Return the `Distribution` selected for Zinc based on execution strategy.
-
-    :rtype: pants.java.distribution.distribution.Distribution
-    """
+  def dist(self) -> Distribution:
+    """Return the `Distribution` selected for Zinc based on execution strategy."""
     underlying_dist = self.underlying_dist
-    if self._execution_strategy != NailgunTaskBase.HERMETIC:
-      # symlink .pants.d/.jdk -> /some/java/home/
-      jdk_home_symlink = os.path.relpath(
-        os.path.join(self._zinc_factory.get_options().pants_workdir, '.jdk'),
-        get_buildroot())
-
-      # Since this code can be run in multi-threading mode due to multiple
-      # zinc workers, we need to make sure the file operations below are atomic.
-      with self._lock:
-        # Create the symlink if it does not exist, or points to a file that doesn't exist,
-        # (e.g., a JDK that is no longer present), or points to the wrong JDK.
-        if (not os.path.exists(jdk_home_symlink) or
-            os.readlink(jdk_home_symlink) != underlying_dist.home):
-          safe_delete(jdk_home_symlink)  # Safe-delete, in case it's a broken symlink.
-          os.symlink(underlying_dist.home, jdk_home_symlink)
-
-      return Distribution(home_path=jdk_home_symlink)
-    else:
+    if self._execution_strategy == NailgunTaskBase.HERMETIC:
       return underlying_dist
+    # symlink .pants.d/.jdk -> /some/java/home/
+    jdk_home_symlink = (
+      Path(self._zinc_factory.get_options().pants_workdir) / '.jdk'
+    ).relative_to(get_buildroot())
+
+    # Since this code can be run in multi-threading mode due to multiple
+    # zinc workers, we need to make sure the file operations below are atomic.
+    with self._lock:
+      # Create the symlink if it does not exist, or points to a file that doesn't exist,
+      # (e.g., a JDK that is no longer present), or points to the wrong JDK.
+      if not jdk_home_symlink.exists() or jdk_home_symlink.resolve() != Path(underlying_dist.home):
+        safe_delete(str(jdk_home_symlink))  # Safe-delete, in case it's a broken symlink.
+        safe_mkdir_for(jdk_home_symlink)
```

Comment:
This line was necessary to get `./pants --no-v1 --v2 test tests/python/pants_test/backend/jvm/tasks/jvm_compile` (note this is local execution). 

When you include `//:build_root` in `TestBase`, if you were to run `Path.cwd()` at this particular line then it no longer results in the actual build root of `/Users/eric/DocsLocal/code/projects/pants` but instead the `/Users/eric/DocsLocal/code/projects/pants/.pants.d/process-executionYmgTGM` folder used by V2, which is what we want. As a result, it tries to create the symlink in `.pants.d/process-executionYmgTGM/.pants.d`, which does not exist. We add this line to ensure that the parent folder does exist, so that the symlink can be created properly.